cluster/service.go

```diff
@@ -54,7 +54,7 @@ type clusterService struct {
 }
 
 // NewClusterService creates an instance of service that using the Auth service retrieves information about clusters
-func NewClusterService(refreshInt time.Duration, authService *auth.Service) Service {
+func NewClusterService(refreshInt time.Duration, authService auth.Service) Service {
```

Comment:
why don't we pass the client options in the constructor here. rather than using `s.authService.GetClientOptions()...` on line 156?

Reply:
because this information is stored in auth service and I'm passing the auth service to the constructor anyway. 
But I see your point - this didn't look so bad in the previous solution - without the interface. Now it looks a bit weird. 
changed here: 3d217d2 
thanks for pointing it out